qemu/tests/qemu_guest_agent.py

```diff
@@ -1723,17 +1729,33 @@ def _change_bl(self, session):
         for windows guest.
         """
         if self.params.get("os_type") == "linux":
-            cmd_black_list = self.params["black_list"]
-            cmd_blacklist_backup = self.params["black_list_backup"]
+            cmd_blacklist_backup = self.params["black_file_backup"]
             session.cmd(cmd_blacklist_backup)
-            for black_cmd in cmd_black_list.split():
-                bl_check_cmd = self.params["black_list_check_cmd"] % black_cmd
-                bl_change_cmd = self.params["black_list_change_cmd"] % black_cmd
+            full_qga_ver = self._get_main_qga_version(session, self.vm, main_ver=False)
+            full_qga_ver = version.parse(full_qga_ver)
+            if full_qga_ver >= version.parse("8.1.0-5"):
+                cmd_black_list = self.params["black_list_new"]
+                bl_check_cmd = self.params["black_list_check_cmd"] % cmd_black_list
+                bl_change_cmd = self.params["black_list_change_cmd"] % cmd_black_list
                 session.cmd(bl_change_cmd)
                 output = session.cmd_output(bl_check_cmd)
-                if not output == "":
+                if output == "":
                     self.test.fail("Failed to change the cmd to "
                                    "white list, the output is %s" % output)
+            else:
+                cmd_black_list = self.params["black_list"]
+                for black_cmd in cmd_black_list.split():
+                    bl_check_cmd = self.params["black_list_check_cmd"] % black_cmd
+                    if (self.params["gagent_check_type"] == "blacklist" and
+                            full_qga_ver >= version.parse("7.2.0-4")):
+                        bl_change_cmd = self.params["black_list_change_cmd"]
+                    else:
+                        bl_change_cmd = self.params["black_list_change_cmd"] % black_cmd
+                    session.cmd(bl_change_cmd)
+                    output = session.cmd_output(bl_check_cmd)
+                    if not output == "":
+                        self.test.fail("Failed to change the cmd to "
+                                       "white list, the output is %s" % output)
```

Comment:
@6-dehan As we discussed, please re-organise this part.

Reply:
okay, I'm on it.

Reply:
I adjusted the logic of this part and merged and simplified the code. here are the testing results:

- 9.1 or less:
 (1/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.seek.q35: PASS (69.48 s)
 (2/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.write.q35: PASS (69.59 s)
 (3/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.read.q35: PASS (70.15 s)
 (4/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_fsfreeze.q35: PASS (69.65 s)
 (5/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (73.69 s)
 (6/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist.q35: PASS (104.40 s)
 (7/7) Host_RHEL.m9.u1.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.isa_serial.gagent_check_log.q35: PASS (70.23 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
- 9.2~9.3:
 (1/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.seek.q35: PASS (53.85 s)
 (2/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.write.q35: PASS (53.15 s)
 (3/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.read.q35: PASS (55.17 s)
 (4/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_fsfreeze.q35: PASS (53.95 s)
 (5/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (62.98 s)
 (6/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist.q35: PASS (53.53 s)
 (7/7) Host_RHEL.m9.u3.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.isa_serial.gagent_check_log.q35: PASS (51.74 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
- 9.4 or above:
 (1/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.seek.q35: PASS (35.86 s)
 (2/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.write.q35: PASS (35.87 s)
 (3/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.read.q35: PASS (36.61 s)
 (4/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_file.with_fsfreeze.q35: PASS (35.47 s)
 (5/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.check_guest_exec.q35: PASS (40.54 s)
 (6/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.virtio_serial.gagent_check_blacklist.q35: PASS (35.21 s)
 (7/7) Host_RHEL.m9.u4.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.4.0.x86_64.io-github-autotest-qemu.qemu_guest_agent.isa_serial.gagent_check_log.q35: PASS (35.38 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0